src/components/App.js

```diff
@@ -359,39 +382,41 @@ export default class App extends React.Component {
      *   - Pause the player because this means all of our urls are having issues
      */
 
-    const { mounts, remotes, erroredStreams, url } = this.state;
-    const sortedStreams = this.sortStreams([...mounts, ...remotes]);
-    const currentStream = sortedStreams.find(stream => stream.url === url);
-    const isStreamInErroredList = erroredStreams.some(
-      stream => stream.url === url
-    );
-    const newErroredStreams = isStreamInErroredList
-      ? erroredStreams
-      : [...erroredStreams, currentStream];
-
-    // Pause if all streams are in the errored list
-    if (newErroredStreams.length === sortedStreams.length) {
-      this.pause();
-      return;
-    }
-
-    // Available streams are those in `sortedStreams`
-    // that don't exist in the errored list
-    const availableStreams = sortedStreams.filter(
-      stream =>
-        !newErroredStreams.some(
-          erroredStream => erroredStream.url === stream.url
-        )
-    );
+    if (this.state.playing) {
+      const { mounts, remotes, erroredStreams, url } = this.state;
+      const sortedStreams = this.sortStreams([...mounts, ...remotes]);
+      const currentStream = sortedStreams.find(stream => stream.url === url);
+      const isStreamInErroredList = erroredStreams.some(
+        stream => stream.url === url
+      );
+      const newErroredStreams = isStreamInErroredList
+        ? erroredStreams
+        : [...erroredStreams, currentStream];
+
+      // Pause if all streams are in the errored list
+      if (newErroredStreams.length === sortedStreams.length) {
+        this.pause();
+        return;
+      }
 
-    // If the url is already in the errored list, use another url
-    if (isStreamInErroredList) {
-      this.setUrl(availableStreams[0].url);
-    } else {
-      // Otherwise, add the url to the errored list, then use another url
-      this.setState({ erroredStreams: newErroredStreams }, () =>
-        this.setUrl(availableStreams[0].url)
+      // Available streams are those in `sortedStreams`
+      // that don't exist in the errored list
+      const availableStreams = sortedStreams.filter(
+        stream =>
+          !newErroredStreams.some(
+            erroredStream => erroredStream.url === stream.url
+          )
       );
+
+      // If the url is already in the errored list, use another url
+      if (isStreamInErroredList) {
+        this.setUrl(availableStreams[0].url);
+      } else {
+        // Otherwise, add the url to the errored list, then use another url
+        this.setState({ erroredStreams: newErroredStreams }, () =>
+          this.setUrl(availableStreams[0].url)
+        );
+      }
```

Comment:
```suggestion
      const availableUrls = sortedStreams.filter(
        stream =>
          !newErroredStreams.some(
            erroredStream => erroredStream.url === stream.url
          )
      ).map(({ url }) => url);

      // If the url is already in the errored list, use another url
      if (isStreamInErroredList) {
        this.setUrl(availableUrls [0]);
      } else {
        // Otherwise, add the url to the errored list, then use another url
        this.setState({ erroredStreams: newErroredStreams }, () =>
          this.setUrl(availableUrls[0])
        );
      }
```
If there are no availableStreams, this will error out.  Would it be better to just get availableUrls instead?

Reply:
That makes sense. Thanks